### ISSUE TITLE
Saved JSON model files cannot be loaded back in.

### ISSUE THREAD

**Issue by ml-l** [vinecopulib/pyvinecopulib#84]
I've very recently updated the version of pyvinecopulib as I saw there was a recent new release and tried (via the current new release) to:

- train a copula, 
- save it in json via the `.to_json()` method of the `Vinecop` class
- and then load it back in via the `Vinecop` class init 

but I get a `[json.exception.type_error.302] type must be array, but is null`.

From looking into a model JSON file that I've saved form a previous version of the library I do notice some JSON structure changes.

With the new library I have also tried loading model files created from both the older and current version of pyvinecopulib and neither work.

**Comment by tvatter** [comment]
As mentioned in the release notes for [vinecopulib 0.6.1](https://github.com/vinecopulib/vinecopulib/releases/tag/v0.6.1), there has been a breaking API change regarding the structure of the saved JSON files. It means that older models can't be loaded with the new version, and newer models can't be loaded with the old one. The changes were done so that the read/write process is faster and the file sizes are smaller.

I just installed the new version in a fresh conda environment and did the following to quickly test whether there was an issue:

```
import pyvinecopulib as pv
import numpy as np

# Specify pair-copulas
bicop = pv.Bicop(pv.BicopFamily.bb1, 90, [3, 2])
pcs = [[bicop, bicop], [bicop]]

# Specify R-vine matrix
mat = np.array([[1, 1, 1], [2, 2, 0], [3, 0, 0]])

# Set-up a vine copula
cop = pv.Vinecop(mat, pcs)

# Write the vine to a file
cop.to_json('test.json')

# Read the model from file
cop2 = pv.Vinecop('test.json')
```

And for me, it worked. I also tried larger models with nonparametric copulas (see my answer at [the end of this thread](https://github.com/vinecopulib/vinecopulib/issues/504)), no problem either.

Any chance that you can create a MWE to reproduce your issue?

**Comment by ml-l** [comment]
> Any chance that you can create a MWE to reproduce your issue?

With pyvincopulib 0.6.1 I can reproduce the problem with the following example:
```Python
import numpy as np
import pyvinecopulib as pv
col_types = ["c","c","c","c","c","c","c","c","c","c","d","d","d","d","d","d","d","d","d","d","d",
             "d","d","d","d","d","d","d","d","d","d","d","d","d","d","d","d","d","d","d","d","d"]

# only 10 continuous cols
unifs = np.random.uniform(size=(100, len(col_types))) # U = F(X)
unifs_with_lag = np.hstack([unifs.copy(), np.zeros((100,len(col_types)-10))]) #copy of data with the U^- = F(X-1) discrete columns
# "discretise" discrete columns (all bernoulli with p=thres_i)
for i in range(10, len(col_types)):
    thres = np.random.uniform() 
    ind = unifs[:,i] <= thres
    unifs[ind,i] = 1
    unifs[~ind,i] = 1-thres
    unifs_with_lag[:,i] = unifs[:,i]
    unifs_with_lag[ind,len(col_types)+i-10] = 1-thres
    unifs_with_lag[~ind,len(col_types)+i-10] = 0

controls = pv.FitControlsVinecop(family_set=[pv.BicopFamily.clayton, pv.BicopFamily.frank, 
            pv.BicopFamily.gaussian, pv.BicopFamily.indep, pv.BicopFamily.student, pv.BicopFamily.gumbel, 
            pv.BicopFamily.joe, pv.BicopFamily.bb1, pv.BicopFamily.bb6, pv.BicopFamily.bb7, 
            pv.BicopFamily.bb8],
            num_threads=8, parametric_method="mle", selection_criterion="aic",show_trace=True)
cop = pv.Vinecop(unifs_with_lag, controls=controls, var_types=col_types)
cop.to_json("./test_save.json")
loaded = pv.Vinecop(filename="./test_save.json") # Error on this line
# RuntimeError: [json.exception.type_error.302] type must be array, but is null
```

**Comment by tvatter** [comment]
Thanks a lot! I found the issue, it was related to the independence bivariate copula, encoded as a json null which couldn't be read properly. It will be fixed in the next release. In the meantime, it is already corrected in the [fix_from_json branch](https://github.com/vinecopulib/pyvinecopulib/tree/fix_from_json). I also added better unit tests at the C++ level to ensure that something like that does not happen again. Apologies!
